utils/michigan.py

```diff
@@ -0,0 +1,1052 @@
+
+        id_mapping = pd.concat([corporations, committees, vendors], ignore_index=True)
+        id_mapping["state"] = "MI"
+        id_mapping["entity_type"] = "Organization"
+
+        id_mapping = id_mapping[self.id_mapping_column_order].copy()
+
+        return id_mapping
+
+    def create_transactions_id_mapping(
+        self,
+        org_com: pd.DataFrame,
+        ind_com: pd.DataFrame,
+        com_vend: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """Creates the ID mapping dataframe for transactions
+
+        Inputs:
+            org_com: dataframe with organizations to committee transactions
+            ind_com: dataframe with individual to committee transactions
+            com_vend: dataframe with  committee  to vendor transactions
+
+        Returns: id_mapping: dataframe in the ID mapping format
+        """
+        org_com = org_com[["year", "transaction_id"]].copy()
+        ind_com = ind_com[["year", "transaction_id"]].copy()
+        com_vend = com_vend[["year", "transaction_id"]].copy()
+
+        org_com = org_com.rename(columns={"transaction_id": "database_id"})
+        ind_com = ind_com.rename(columns={"transaction_id": "database_id"})
+        com_vend = com_vend.rename(columns={"transaction_id": "database_id"})
+
+        id_mapping = pd.concat([org_com, ind_com, com_vend], ignore_index=True)
+        id_mapping["provided_id"] = np.nan
+        id_mapping["state"] = "MI"
+        id_mapping["entity_type"] = "Transaction"
+
+        id_mapping = id_mapping[self.id_mapping_column_order].copy()
+
+        return id_mapping
+
+    # NOTE: universal helper functions for creating the tables are below
+
+    def filter_dataframe(
+        self, merged_campaign_dataframe: pd.DataFrame, column_name: str
+    ):
+        """Filters the inputted dataframe based on the column name
+
+        Inputs:
+            merged_campaign_dataframe:
+            column_name: name of a column to filter the dataset on
+
+        Returns:
+            filtered_df: dataframe filtered based on the given column name
+
+        """
+        filtered_df = merged_campaign_dataframe[
+            merged_campaign_dataframe[column_name].notnull()
+        ]
+        # returns all the columns and rows associated with the column name
+        # that are not null
+
+        return filtered_df
+
+    def filter_irrelevant_columns(
+        self, filtered_dataframe: pd.DataFrame, column_lst: list[str]
+    ) -> pd.DataFrame:
+        """Filters the dataframe by only keeping the columns listed
+
+        Inputs:
+            filtered_dataframe: dataframe of filtered campaign finance data
+            column_lst: list of strings of column names
+
+        Returns:
+            filtered_dataframe: filtered dataframe with only the columns
+            listed above
+        """
+        filtered_dataframe = filtered_dataframe[column_lst].copy()
+
+        return filtered_dataframe
+
+    def reorder_columns(
```

Comment:
I also don't think this is a necessary helper function

Reply:
same comment as above: The same piece of code is used in multiple places. Do you want me to get rid of it?